src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -660,101 +661,130 @@ public long GetNumberOfCounterEntries(DocumentsOperationContext context)
             return fst.NumberOfEntries;
         }
 
-        public void UpdateDocumentCounters(DocumentsOperationContext context, BlittableJsonReaderObject doc, string docId, BlittableJsonReaderObject metadata, List<CounterOperation> countersOperations)
+        public void UpdateDocumentCounters(DocumentsOperationContext context, BlittableJsonReaderObject doc, string docId, List<CounterOperation> countersOperations)
         {
-            List<string> updates = null;
-            if (metadata.TryGet(Constants.Documents.Metadata.Counters, out BlittableJsonReaderArray counters))
+            BlittableJsonReaderArray metadataCounters = null;
+            var initCounterListSize = countersOperations.Count;
+            if (doc.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) &&
+                metadata.TryGet(Constants.Documents.Metadata.Counters, out metadataCounters))
             {
-                foreach (var operation in countersOperations)
+                initCounterListSize += metadataCounters.Length;
+            }
+
+            //Todo Maybe should not read from the table if there is metadata on document
```

Comment:
We can't assume that the metadata value is correct, so we have to read from the table anyway